src/reference/asciidoc/hazelcast.adoc

```diff
@@ -0,0 +1,600 @@
+It is mandatory attribute;
+* `cache`: Specifies distributed `IMap` reference which is queried.
+It is mandatory attribute;
+* `iteration-type`: Specifies result type.
+Distributed SQL can be run on `EntrySet`, `KeySet`, `LocalKeySet` or `Values`.
+It is optional attribute with `VALUE` by default.
+Supported values are `ENTRY, `KEY`, `LOCAL_KEY` and `VALUE`;
+* `distributed-sql`: Specifies where clause of sql statement.
+It is mandatory attribute.
+
+[[hazelcast-outbound-channel-adapter]]
+=== Hazelcast Outbound Channel Adapter
+
+Hazelcast Outbound Channel Adapter listens its defined channel and writes incoming messages to related distributed cache.
+It expects one of `cache`, `cache-expression` or `HazelcastHeaders.CACHE_NAME` for distributed object definition.
+Supported Distributed Objects are: `IMap`, `MultiMap`, `ReplicatedMap`, `IList`, `ISet`, `IQueue` and `ITopic`.
+
+====
+[source, java, role="primary"]
+.Java
+----
+@Bean
+public MessageChannel distributedMapChannel() {
+    return new DirectChannel();
+}
+
+@Bean
+public IMap<Integer, String> distributedMap() {
+    return hzInstance().getMap("Distributed_Map");
+}
+
+@Bean
+public HazelcastInstance hzInstance() {
+    return Hazelcast.newHazelcastInstance();
+}
+
+@Bean
+@ServiceActivator(inputChannel = "distributedMapChannel")
+public HazelcastCacheWritingMessageHandler hazelcastCacheWritingMessageHandler() {
+    HazelcastCacheWritingMessageHandler handler = new HazelcastCacheWritingMessageHandler();
+    handler.setDistributedObject(distributedMap());
+    handler.setKeyExpression(new SpelExpressionParser().parseExpression("payload.id"));
+    handler.setExtractPayload(true);
+    return handler;
+}
+----
+[source, xml, role="secondary"]
+.XML
+----
+<int-hazelcast:outbound-channel-adapter channel="mapChannel"
+                    cache-expression="headers['CACHE_HEADER']"
+                    key-expression="payload.key"
+                    extract-payload="true"/>
+----
+====
+
+It requires the following attributes :
+
+* `channel`: Specifies channel which message is sent;
+* `cache`: Specifies distributed object reference.
+Optional;
+* `cache-expression`: Specifies distributed object via Spring Expression Language(SpEL).
+Optional;
+* `key-expression`: Specifies key of key-value pair via Spring Expression Language(SpEL).
+Optional and required for only for `IMap`, `MultiMap` and `ReplicatedMap` distributed data structures.
+* `extract-payload`: Specifies whole message or just payload to send.
+Optional attribute with `true` by default.
+If it is true, just payload will be written to distributed object.
+Otherwise, whole message will be written by covering both message header and payload.
+
+By setting distributed object name in the header, messages can be written to different distributed objects via same channel.
+If `cache` or `cache-expression` attributes are not defined, a `HazelcastHeaders.CACHE_NAME` header has to be set in a request `Message`.
```

Comment:
```suggestion
It requires the following attributes :

* `channel`: Specifies the channel to which messages are sent;
* `cache`: Specifies the distributed object reference.
Optional;
* `cache-expression`: Specifies the distributed object via Spring Expression Language (SpEL).
Optional;
* `key-expression`: Specifies the key of a key-value pair via Spring Expression Language (SpEL).
Optional and required for only for `IMap`, `MultiMap` and `ReplicatedMap` distributed data structures.
* `extract-payload`: Specifies whether to send the whole message or just the payload.
Optional attribute with `true` being the default.
If it is true, just the payload will be written to the distributed object.
Otherwise, the whole message will be written by converting both message headers and payload.

By setting distributed object name in the header, messages can be written to different distributed objects via same channel.
If `cache` or `cache-expression` attributes are not defined, a `HazelcastHeaders.CACHE_NAME` header has to be set in a request `Message`.
```